scala-package/infer/src/main/scala/org/apache/mxnet/infer/javaapi/Predictor.scala

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.mxnet.infer.javaapi
+
+import org.apache.mxnet.javaapi.{Context, DataDesc, NDArray}
+
+import scala.collection.JavaConverters
+import scala.collection.JavaConverters._
+
+/**
+  * Implementation of prediction routines.
+  *
+  * @param modelPathPrefix     Path prefix from where to load the model artifacts.
+  *                            These include the symbol, parameters, and synset.txt
+  *                            Example: file://model-dir/resnet-152 (containing
+  *                            resnet-152-symbol.json, resnet-152-0000.params, and synset.txt).
+  * @param inputDescriptors    Descriptors defining the input node names, shape,
+  *                            layout and type parameters
+  *                            <p>Note: If the input Descriptors is missing batchSize
+  *                            ('N' in layout), a batchSize of 1 is assumed for the model.
+  * @param contexts            Device contexts on which you want to run inference; defaults to CPU
+  * @param epoch               Model epoch to load; defaults to 0
+
+  */
+
+// JavaDoc description of class to be updated in https://issues.apache.org/jira/browse/MXNET-1178
+class Predictor(val predictor: org.apache.mxnet.infer.Predictor){
```

Comment:
constructor private? Please make all the constructors that uses the Scala Object as private